Makefile

```diff
@@ -28,13 +28,19 @@ docker:
 clean:
 	@$(GO) clean
 
+stop-e2e:
+	@docker compose -f tests/e2e/docker-compose.e2e.yml down -v -t 0
+	rm -rf tests/e2e/testnet/net-runner
+
 test:
 	@$(GO) test -short -v ./...
 
 test-ci:
 	@$(GO) test -race -coverpkg=./... -coverprofile=coverage.txt -covermode=atomic -v ./...
 
-test-e2e: export OASIS_INDEXER_E2E = true 
+# Run the e2e tests locally, assuming the environment is already set up.
+# The recommended usage is via the `start-e2e` target which sets up the environment first.
+test-e2e: export OASIS_INDEXER_E2E = true
```

Comment:
is this a duplicate rule?

Reply:
Oh line 44 also defines a `test-e2e` rule, which will overwrite this one. You probably want something like 
```
OASIS_INDEXER_E2E=true @$(GO) test -race -coverpkg=./... -coverprofile=coverage.txt -covermode=atomic -v ./tests/e2e
```

Reply:
Uhhh, somehow the line 44 version works and the below doesn't?

```
OASIS_INDEXER_E2E=true @$(GO) test -race -coverpkg=./... -coverprofile=coverage.txt -covermode=atomic -v ./tests/e2e
```

Reply:
Line 44 doesn't redefine/overwrite the rule because line 43 is not a rule, it's more like an annotation of the rule. This is how you export env variables to make targets. See [official docs](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html) and [a more readable discussion](https://stackoverflow.com/questions/23843106/how-to-set-child-process-environment-variable-in-makefile)

The line in the comment just above mine (assuming you tried it verbatim) doesn't work because the `@` is Makefile syntax for "don't log this command just before you execute it"; you want `@OASIS_INDEXER_E2E=true $(GO) test [...]`

I'm fine with either syntax :)

Reply:
Thanks. I learned something new.